benches/functional.rs

```diff
@@ -0,0 +1,101 @@
+use bst::functional::Tree;
+use criterion::{
+    black_box, criterion_group, criterion_main, AxisScale, BenchmarkId, Criterion,
+    PlotConfiguration,
+};
+
+/// Returns how many nodes are needed to fill a binary tree with `num_levels` levels.
+fn num_nodes_in_full_tree(num_levels: usize) -> usize {
+    2usize.pow(num_levels as u32) - 1
+}
+
+/// Builds a tree by inserting values in an unbalanced manner. This adds elements in an
+/// ascending manner to ensure the tree is unbalanced (unless it is self-balancing!).
+fn get_unbalanced_tree(num_levels: usize) -> Tree<i32, i32> {
+    let mut tree = Tree::new();
+    let tree_size = num_nodes_in_full_tree(num_levels);
+    for x in (0..).take(tree_size) {
```

Comment:
Note to self - this is how to use half-open ranges, with `(0..)`

Reply:
question - would `(0..tree_size)` not work?

Reply:
It totally would work!